librz/asm/arch/evm/evmasm.c

```diff
@@ -0,0 +1,638 @@
+// SPDX-FileCopyrightText: 2023 gogo <gogo246475@gmail.com>
+// SPDX-FileCopyrightText: 2012-2020 pancake <pancake@nopcode.org>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_util.h>
+#include <rz_types.h>
+#include <rz_asm.h>
+#include <string.h>
+
+static void str_op(char *c) {
+	if ((c[0] <= 'Z') && (c[0] >= 'A')) {
+		c[0] += 0x20;
+	}
+}
+
+static int gb_reg_idx(char r) {
+	const char *rstr = "bcdehl a";
+	const char *ptr = strchr(rstr, r);
+	return ptr ? (int)(size_t)(ptr - rstr) : -1;
+}
+
+static bool gb_parse_cb1(ut8 *buf, const int minlen, char *buf_asm, ut8 base) {
+	// minlen varries between 4 and 6
+	int i;
+	size_t j;
+	if (strlen(buf_asm) < minlen || minlen < 1) {
+		return false;
+	}
+	buf[0] = base;
+	char *ptr_asm = buf_asm + minlen - 1;
+	j = strlen(ptr_asm);
+	rz_str_replace_in(ptr_asm, (ut32)j, "[ ", "[", true);
+	rz_str_replace_in(ptr_asm, (ut32)j, " ]", "]", true);
+	rz_str_do_until_token(str_op, buf_asm, ' ');
+	i = gb_reg_idx(buf_asm[minlen - 1]);
+	if (i != (-1)) {
```

Comment:
this code is from the gb. why is there? EVM does not use the same assembly as the gb.
https://github.com/rizinorg/rizin/blob/dev/librz/asm/arch/gb/gbasm.c